helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -688,6 +738,21 @@ private GenericHelixController(PipelineRegistry registry, PipelineRegistry taskR
       _taskEventThread = null;
     }
 
+    if (_enabledPipelineTypes.contains(Pipeline.Type.MANAGEMENT_MODE)) {
```

Comment:
The code her is exact same logic with previous two, just variable different. Can we have a private function to abstract the common logic and make the code more clear?

Reply:
Added TODO: refactor to simplify below similar code of the 3 pipelines.
This should be wrapped into an object and use a HashMap to map each pipeline to its objects(thread, queue, etc).